ui/src/main/java/org/pentaho/di/ui/spoon/tree/TreeManager.java

```diff
@@ -188,6 +190,11 @@ private <T> void render( TreeNode treeNode, T tree ) {
     TreeItem childTreeItem = createTreeItem( treeNode, tree );
     if ( treeNode.hasChildren() ) {
       for ( TreeNode childTreeNode : treeNode.getChildren() ) {
+        Repository repository = Spoon.getInstance().getRepository();
+        if( childTreeNode.getLabel().equals( "VFS Connections" ) && repository!=null
+          && repository.getUserInfo()!=null && !repository.getUserInfo().isAdmin() ){
+          continue;
+        }
```

Comment:
Most likely the change will be in java, so a unit test will be in order.

Reply:
holding on to the test case till re-review is complete for acceptance of location of code